lib/pure/times.nim

```diff
@@ -1116,7 +1127,7 @@ when not defined(JS):
         gmtoff {.importc: "tm_gmtoff".}: clong
   else:
     type
-      StructTM {.importc: "struct tm", final.} = object
+      StructTM {.importc: "struct tm", pure, final.} = object
```

Comment:
>  but if I create a random Nim type without importc and without pure, it will not be abi compatible with c - it will have a type header..

It really shouldn't have a type header. Are you sure?